### DISCUSSION TITLE
break-words doesn't work

### DISCUSSION THREAD

**Discussion by gurgunday** [tailwindlabs/tailwindcss#10545] [Help] [answered]
Hey!

So I have an issue where `break-words` doesn't really break a word when needed. Instead, it is the `overflow-wrap: anywhere;` rule that breaks a single word if necessary.

But Tailwind doesn't support it. Maybe there should be a separate class for this CSS rule?

Here's a comparison:

## `break-words`
<img width="382" alt="Screenshot 2023-02-09 at 16 43 26" src="https://user-images.githubusercontent.com/74139498/217862762-5bde268b-2a03-4804-8308-d1c5b044959d.png">

## `overflow-wrap: anywhere;`
<img width="371" alt="Screenshot 2023-02-09 at 16 43 37" src="https://user-images.githubusercontent.com/74139498/217862773-94ab24fd-d0df-43f5-b4f6-6faf975b6c22.png">

**Comment by adamwathan** [comment, marked as answer]
Can you share a code example? I can't seem to reproduce that quickly here:

https://play.tailwindcss.com/NVaJdxhao6

**Reply by gurgunday** [reply]
Here you go! Sorry for the weird code 🤣 pulled it out of [HEYHEY](https://heyhey.to/G).

https://play.tailwindcss.com/j8z6dsa8Yz

If I make the screen size smaller than the width of the `max-width`, the text just breaks out of the container and I have no idea why.

<img width="561" alt="Screenshot 2023-02-09 at 18 22 14" src="https://user-images.githubusercontent.com/74139498/217891807-eef0e8c8-3a0d-44f8-a6a3-53c1cb263916.png">

Now, if `overflow-wrap: anywhere;` is added to the style attribute of the div that has the id `here`, it works exactly as expected.

Here's the same code with `overflow-wrap: anywhere;`:
<img width="704" alt="Screenshot 2023-02-09 at 18 29 42" src="https://user-images.githubusercontent.com/74139498/217891754-b8de662c-6df9-4fa2-a328-68ca29fd0e3b.png">

**Reply by wongjn** [reply]
You could add `w-full` to `#here` to enforce the width of it: https://play.tailwindcss.com/TjEwiDQPma?size=102x720

**Reply by gurgunday** [reply]
It worked! Thanks a lot, but how about this one? I think flex is what causes the issue here:

https://play.tailwindcss.com/WUAXigE8jn?size=494x720

This one is similar if you don't want to mess with DaisyUI:

https://play.tailwindcss.com/2PmTquhbiO?size=358x720

`overflow-wrap: anywhere;` seems to work both with flex and normal containers.

**Reply by adamwathan** [reply]
Yep discovered the same as @wongjn — `w-full` seems to fix it, as does removing `flex` from the parent. So this is one of those weird flexbox + intrinsic-width issues that come up all the time I think.

**Reply by adamwathan** [reply]
@gurgunday So the second example is definitely the classic "`min-w-0` fixes it" example — here you go:

https://play.tailwindcss.com/QBmgJ896Ho?size=358x720

A bit annoying because you need to wrap the text in a span to actually be able to target it, but the underlying issue is that text elements have a default min-width value based on their content, and this interacts with how shrinking/basis works in flexbox in surprising ways sometimes. Basically you need to give flexbox "permission" to try and compress the text with `min-w-0` and then the wrapping will happen as it would in a non-flex container.

**Reply by gurgunday** [reply]
> So the second example is definitely the classic "min-w-0 fixes it" example — here you go:

🤯

> Yep discovered the same as @wongjn — `w-full` seems to fix it, as does removing `flex` from the parent. So this is one of those weird flexbox + intrinsic-width issues that come up all the time I think.

@adamwathan, no expert here, but as I said, all the inconsistencies seem to go away if `break-words` resolves to `overflow-wrap: anywhere;`. Would you consider that change?

**Reply by adamwathan** [reply]
I think we should add support for that value either way for sure yeah, not sure if it should replace the current one though. I think it makes more sense to just support everything, but we probably want to rethink these APIs because they are starting to feel a bit inconsistent and disconnected from the underlying CSS.

You can also use arbitrary properties to do this in the mean time:

`<div class=“[overflow-wrap:anywhere]”>`

**Reply by kasvith** [reply]
this reply saved me

**Reply by Bennyjoez** [reply]
Your suggestion worked for me. Thank you... 💪🏾  @adamwathan

**Reply by kaiwen-wang** [reply]
arbitrary properties worked

**Reply by filip-rybczynski** [reply]
Absolute lifesaver from @adamwathan, thanks so much

**Reply by hampak** [reply]
This reply saved me, thank you so much! @adamwathan

**Comment by Hairraiserz** [comment]
Thanks for sharing this, this is amazing. i like it. [salon in Chandigarh](https://hairraiserzluxesunny125.business.site/)
